packages/web-frontend/src/containers/OverlayDiv/components/ProjectPage/ProjectCardList.js

```diff
@@ -5,26 +5,51 @@
 
 import React from 'react';
 import { ProjectCard } from './ProjectCard';
+import {
+  ALLOWED_PROJECT_ACCESS_TYPE,
+  DENIED_PROJECT_ACCESS_TYPE,
+  PENDING_PROJECT_ACCESS_TYPE,
+} from '../../../../constants';
 
 const EXPLORE_CODE = 'explore';
 
-export const ProjectCardList = ({ projects, accessType, action, actionText }) => {
-  const hasPendingType = accessType === 'pending';
-  const hasAccessType = hasPendingType ? false : accessType;
-  return projects
-    .filter(
-      ({ code, hasAccess, hasPendingAccess = false }) =>
-        code !== EXPLORE_CODE && hasAccess === hasAccessType && hasPendingAccess === hasPendingType,
-    )
-    .sort((a, b) => a.sortOrder - b.sortOrder)
-    .map(project => (
-      <ProjectCard
-        key={project.name}
-        projectAction={() => action(project)}
-        actionText={actionText}
-        accessType={hasAccessType}
-        hasAccessPending={hasPendingType}
-        {...project}
-      />
-    ));
+export const ProjectCardList = ({ projects, actions, isUserLoggedIn }) => {
+  const actionText = {
+    [PENDING_PROJECT_ACCESS_TYPE]: 'Approval in progress',
+    [ALLOWED_PROJECT_ACCESS_TYPE]: 'View project',
+    [DENIED_PROJECT_ACCESS_TYPE]: isUserLoggedIn ? 'Request access' : 'Log in',
+  };
+  const sortedProjects = [
+    ALLOWED_PROJECT_ACCESS_TYPE,
+    PENDING_PROJECT_ACCESS_TYPE,
+    DENIED_PROJECT_ACCESS_TYPE,
+  ].reduce((result, accessType) => {
+    const action = actions[accessType];
+    const accessTypeProjects = projects.filter(({ code, hasAccess, hasPendingAccess = false }) => {
+      let projectAccessType = DENIED_PROJECT_ACCESS_TYPE;
+      if (hasAccess) projectAccessType = ALLOWED_PROJECT_ACCESS_TYPE;
+      if (hasPendingAccess) projectAccessType = PENDING_PROJECT_ACCESS_TYPE;
+      return code !== EXPLORE_CODE && projectAccessType === accessType;
+    });
+    return [
+      ...result,
+      ...accessTypeProjects
+        .sort((a, b) => a.sortOrder - b.sortOrder)
+        .map(project => ({
+          ...project,
+          action,
+          actionText: actionText[accessType],
+          accessType,
+        })),
+    ];
+  }, []);
+  return sortedProjects.map(project => (
+    <ProjectCard
+      key={project.name}
+      projectAction={() => project.action(project)}
+      actionText={project.actionText}
+      accessType={project.accessType}
+      {...project}
```

Comment:
There is a convention that Edwin likes where we list out the object properties so that it's more transparent about what data is getting passed in. There are exceptions where we don't do it if we are extending a component and would have to list lots of props just to add one new one but here it seems like a good idea.
```suggestion
 name={project.name}
 etc
```